robotoff/workers/tasks/import_image.py

```diff
@@ -126,57 +205,155 @@ def save_image(
     return image_model
 
 
-def run_object_detection(
-    barcode: str, image: Image.Image, source_image: str, server_domain: str
+def run_nutrition_table_object_detection(
+    barcode: str, image_url: str, server_domain: str
 ):
+    logger.info(
+        f"Running nutrition table object detection for product {barcode} "
+        f"({server_domain}), image {image_url}"
+    )
+
+    image = get_image_from_url(image_url, error_raise=False, session=http_session)
+
+    if image is None:
+        logger.info(f"Error while downloading image {image_url}")
+        return
+
+    source_image = get_source_from_url(image_url)
+
+    with db:
+        run_object_detection_model(
+            ObjectDetectionModel.nutrition_table, image, source_image
+        )
+
+
+NUTRISCORE_LABELS = {
+    "nutriscore-a": "en:nutriscore-grade-a",
+    "nutriscore-b": "en:nutriscore-grade-b",
+    "nutriscore-c": "en:nutriscore-grade-c",
+    "nutriscore-d": "en:nutriscore-grade-d",
+    "nutriscore-e": "en:nutriscore-grade-e",
+}
+
+
+def run_nutriscore_object_detection(barcode: str, image_url: str, server_domain: str):
+    logger.info(
+        f"Running nutriscore object detection for product {barcode} "
+        f"({server_domain}), image {image_url}"
+    )
+
+    image = get_image_from_url(image_url, error_raise=False, session=http_session)
+
+    if image is None:
+        logger.info(f"Error while downloading image {image_url}")
+        return
+
+    source_image = get_source_from_url(image_url)
+
+    with db:
+        image_prediction = run_object_detection_model(
+            ObjectDetectionModel.nutriscore, image, source_image
+        )
+
+        if not image_prediction:
+            return
+
+        results = [
+            item for item in image_prediction.data["objects"] if item["score"] >= 0.5
+        ]
+
+        if len(results) > 1:
+            logger.info("more than one nutriscore detected, discarding detections")
+            return
+
+        result = results[0]
+        score = result["score"]
+        label_tag = NUTRISCORE_LABELS[result["label"]]
```

Comment:
Why put it in the `with db`, it shan't it be outside ?

Reply:
`run_object_detection_model` creates a ImagePrediction instance, so we need the DB connection.
But it's indeed better to split it in 2, insight import failure shouldn't mean the ImagePrediction is not imported.